init.go

```diff
@@ -67,7 +41,78 @@ func opensslInit(file string) (major, minor, patch uint, err error) {
 			_OPENSSL_INIT_LOAD_CONFIG|
 			_OPENSSL_INIT_LOAD_CRYPTO_STRINGS,
 		nil) != 1 {
-		return 0, 0, 0, fail("openssl: init crypto")
+		return fail("init crypto")
+	}
+	return nil
+}
+
+// initForCheckVersion loads and initialize only the
+// functions required in [CheckVersion].
+// It returns a function that must be called to release the resources.
+func initForCheckVersion(file string) (func(), error) {
+	// Load the OpenSSL shared library using dlopen.
+	handle, err := openLibrary(file)
+	if err != nil {
+		return nil, err
+	}
+
+	switch vMajor {
+	case 1:
+		mkcgoLoad_init_1(handle)
+	case 3:
+		mkcgoLoad_init_3(handle)
+	default:
+		panic(errUnsupportedVersion())
```

Comment:
Hmm, I think now that the other check for 1 or 3 is further away in `openLibrary`, this panic might actually make it a bit less obvious whether the caller will receive an error or a panic if the version isn't supported. An explicit comment might be reasonable:

```suggestion
		// We shouldn't get here: openLibrary should have already returned an error.
		panic(errUnsupportedVersion())
```